user-interface/src/main/java/life/qbic/datamanager/views/notifications/toasts/Toast.java

```diff
@@ -0,0 +1,259 @@
+package life.qbic.datamanager.views.notifications.toasts;
+
+import static java.util.Objects.nonNull;
+import static java.util.Objects.requireNonNull;
+
+import com.vaadin.flow.component.AttachEvent;
+import com.vaadin.flow.component.Component;
+import com.vaadin.flow.component.ComponentEventListener;
+import com.vaadin.flow.component.UI;
+import com.vaadin.flow.component.button.Button;
+import com.vaadin.flow.component.button.ButtonVariant;
+import com.vaadin.flow.component.html.Div;
+import com.vaadin.flow.component.notification.Notification;
+import com.vaadin.flow.router.BeforeLeaveListener;
+import com.vaadin.flow.router.RouteParameters;
+import com.vaadin.flow.router.RouterLink;
+import com.vaadin.flow.shared.Registration;
+import com.vaadin.flow.theme.lumo.LumoIcon;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+import life.qbic.datamanager.views.general.ComponentFunctions;
+
+/**
+ * A toast notification is a non-modal, unobtrusive window element. It is meant to show information
+ * in brief and auto-expiring windows to the user.
+ * <p>
+ * It is not meant to require user interaction but may provide optional ways for the user to
+ * interact with it.
+ *
+ * @since 1.4.0
+ */
+public final class Toast extends Notification {
+
+  private static final Position DEFAULT_POSITION = Position.BOTTOM_START;
+  private static final boolean DEFAULT_CLOSEABLE = true;
+  private static final boolean DEFAULT_CLOSE_ON_NAVIGATION = true;
+  private static final Level DEFAULT_LEVEL = Level.INFO;
+  private static final int DEFAULT_OPEN_DURATION = 5_000;
+
+  private final List<Registration> closeOnNavigationListeners = new ArrayList<>();
+  private final Button closeButton = closeButton(this);
+
+  private Component content;
+  private Level level;
+  private boolean closeable;
+
+  enum Level {
+    SUCCESS,
+    INFO;
+  }
+
+  static class ToastBuilder {
+
+    Toast build() {
+      return new Toast();
+    }
+
+  }
+
+  private Toast() {
+    super();
+    addClassName("toast-notification");
+    add(closeButton);
+    setCloseable(DEFAULT_CLOSEABLE);
+    setPosition(DEFAULT_POSITION);
+    setType(DEFAULT_LEVEL);
+    closeOnNavigation(DEFAULT_CLOSE_ON_NAVIGATION);
+    setDuration(DEFAULT_OPEN_DURATION);
+  }
+
+  private static Button closeButton(Toast toast) {
```

Comment:
I have a hunch why you did it like this, however this method does some secret action, which is not clear from the signature: it already adds a click listener to the button to close the toast.

I suggest to make that behavioural change more explicit in an own call in the constructor.

Reply:
I think that the behaviour (closing a toast) is inherent to the close button. The close button would not be a close button without the closing behaviour. So this needs to be together imo. I renamed the method to make it more clear.